src/Microsoft.TestPlatform.CrossPlatEngine/Client/ProxyOperationManager.cs

```diff
@@ -78,24 +80,41 @@ public virtual void SetupChannel(IEnumerable<string> sources)
                 // TODO: Fix the environment variables usage
                 var testHostStartInfo = this.testHostManager.GetTestHostProcessStartInfo(sources, null, connectionInfo);
 
+                // Monitor testhost exit.
+                testHostStartInfo.ExitCallback = (process) =>
+                {
+                    if (process.ExitCode != 0)
```

Comment:
Added unit to check weather `ExitCallback` is set or not. For invocation added acceptance test.